wm-system/src/test/kotlin/com/ajaxproject/warehouse/controller/nats/product/CreateProductNatsControllerTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.ajaxproject.warehouse.controller.nats.product
+
+import com.ajaxproject.api.internal.warehousesvc.NatsSubject.Product.CREATE
+import com.ajaxproject.api.internal.warehousesvc.input.reqreply.product.CreateProductRequest
+import com.ajaxproject.api.internal.warehousesvc.input.reqreply.product.CreateProductResponse
+import io.nats.client.Connection
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import java.time.Duration
+
+@SpringBootTest
+@ActiveProfiles("local")
+class CreateProductNatsControllerTest {
+
+    @Autowired
+    lateinit var connection: Connection
+
+    @Test
+    fun testCreatesProductFromValidRequest() {
+        // GIVEN
+        val createProductRequest = CreateProductRequest.newBuilder()
+            .setTitle("Test create product")
+            .setPrice(19.99)
+            .setAmount(200)
+            .setAbout("New test product")
+            .build()
```

Comment:
Please refactor whenever you use the builder pattern in all cases
```suggestion
val createProductRequest = CreateProductRequest.newBuilder().apply {
            title = "Test create product"
            price = 19.99
            amount = 200
            about = "New test product"
        }.build()
```

Reply:
done